stonesoup/predictor/kalman.py

```diff
@@ -376,3 +393,78 @@ def predict(self, prior, timestamp=None, **kwargs):
 
         # and return a Gaussian state based on these parameters
         return GaussianStatePrediction(x_pred, p_pred, timestamp=timestamp)
+
+
+class SqrtKalmanPredictor(KalmanPredictor):
+    r"""The version of the Kalman predictor that operates on the square root parameterisation of
+    the Gaussian state, :class:`~.SqrtGaussianState`.
+
+    The prediction is undertaken in one of two ways. The default is to work in exactly the same
+    way as the parent class, with the exception exception that the predicted covariance is
+    subject to a Cholesky factorisation prior to initialisation of the :class:`~.SqrtGaussianState`
+    output. The alternative, accessible via the :attr:`qr_method = True` flag, is to predict via a
+    modified Gram-Schmidt process. See [1] for details.
+
+    If transition and control models are possessed of the square root form of the covariance (as
+    :attr:`sqrt_covar` in the case of the transition model and :attr:`sqrt_control_noi` for
```

Comment:
I'd be in favour of having a proper look at control models as a separate issue. 

They (control models) exist because they have to in the general case of the Kalman predictors, but they're largely untested and, as you've pointed out, their analogy with other models is a little tenuous. I think more eyes on the issue would help, as well as considering uses they might have outside the Kalman family.